indexclient/client.py

```diff
@@ -192,6 +192,61 @@ def create_alias(
         resp = self._put(url, headers=headers, data=data, auth=self.auth)
         return resp.json()
 
+    def get_latest_revision(self, did):
+        # type: (str) -> Document | None
+        doc = self._get("index", did, "latest").json()
+
+        if doc and "did" in doc:
+            return Document(self, doc["did"], doc)
+        return None
+
+    def auto_add_revision(self, did, version_incr_func=None):
+        """
+        Given a document id,
+            * retrieve the latest version
+            * increment the version if function is specified
+                * using specified function
+            * create a revision
+
+        Thr function should have the following signature
+            * func(str) -> str
+        The use of this function is to cater for cases where the version numbering is
+        complex
+        :type did: str
+        :param did: document id of the index to be revised
+        :type version_incr_func: function[str] -> str
+        :param version_incr_func: if not function is specified, it simply increments the latest version by 1
```

Comment:
where does this happen?

Reply:
When you want to update the version number whenever you add a new version of an entry and you don't want to change the fields. To be frank, even in GDC I don;t see this ever happening, so I'll yank it off